formspree/forms/views.py

```diff
@@ -387,6 +387,14 @@ def form_toggle():
     hashid = request.form.get('hashid')
     form = Form.get_with_hashid(hashid)
 
+    # check that this request came from user dashboard to prevent XSS and CSRF
+    referrer = referrer_to_baseurl(flask.request.referrer)
+    service = referrer_to_baseurl(settings.SERVICE_URL)
+    if referrer != service:
```

Comment:
I'm just saying here that see if the referrer is from the formspree.io domain. It's messy to check the exact URL because there may be arguments attached to the end sometimes and `/dashboard` and `/forms` are actually the same thing, but any request from formspree.io (or the SERVICE_URL) should be OK.